odl/operator/operator.py

```diff
@@ -206,29 +395,64 @@ class Operator(with_metaclass(_OperatorMeta, object)):
     implementation of the respective other is provided.
     """
 
+    def __new__(cls, *args, **kwargs):
+        """Create a new instance."""
+        instance = super().__new__(cls)
+
+        call_has_out, call_out_optional, _ = _dispatch_call_args(cls)
```

Comment:
Doesnt this make creation of operators MUCH slower than before? Have you benchmarked it?

No possibility in not doing this for each and every instance created, looks like a singleton like pattern could work? Perhaps as a metaclass?